File: Daily Challenge/C++/Palindrome Pairs 10.5.2025.cpp
#include <iostream>
#include <vector>
using namespace std;


bool isPalindrome(string s) {
    int len = s.size();
    for (int i = 0; i < len / 2; i++) {
        if (s[i] != s[len - i - 1])
            return false;
    }
    return true;
}

int main() {
    int n;
    cin >> n;
    vector<string> words(n);
    for (int i = 0; i < n; i++) {
        cin >> words[i];
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i != j) {
                string combined = words[i] + words[j];
                if (isPalindrome(combined)) {
                    cout << combined << endl;
                    return 0;
                }
            }
        }
    }
    cout << "-1\n";
    return 0;
}

File: Daily Challenge/Java/Palindrome Pairs 10.5.2025.java
import java.util.Scanner;

public class Main {
    public static boolean isPalindrome(String s) {
        int len = s.length();
        for (int i = 0; i < len / 2; i++) {
            if (s.charAt(i) != s.charAt(len - i - 1))
                return false;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] words = new String[n];
        for (int i = 0; i < n; i++) {
            words[i] = sc.next();
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i != j) {
                    String combined = words[i] + words[j];
                    if (isPalindrome(combined)) {
                        System.out.println(combined);
                        sc.close();
                        return;
                    }
                }
            }
        }
        System.out.println("-1");
        sc.close();
    }
}